src/haddock/libs/libplots.py

```diff
@@ -706,36 +832,129 @@ def clt_table_handler(clt_file, ss_file):
-                values=df.transpose().values.tolist(), align="left", height=40
-                ),
-            )
-        fig.add_trace(table, row=i + 1, col=1)
-    fig.update_layout(title_text="Summary", height=600)
-    return fig
+
+    # The header of the table would be the cluster rank instead of id
+    df_merged = df_merged.set_index("Cluster Rank")
+    table = _pandas_df_to_html_table(df_merged.T, table_id="table")
+    return [table]
+
+
+def _css_styles_for_report():
+    custom_css = '''
+    .table {
+        font-family: Arial, sans-serif;
+        border-collapse: collapse;
+        width: 100%;
+        font-size: 16px;
+        height: auto;
+        overflow-y: auto;
+        }
+    .table th {
+        font-weight: bold;
+        background-color: #f2f2f2;
+        padding: 8px;
+        border: 1px solid #ddd;
+        text-align: left;
+        height: 10px;
+        position: sticky;
+        top: 0;
+        z-index: 1;
+        }
+    .table td {
+        border: 1px solid #ddd;
+        padding: 8px;
+        text-align: left;
+        height: 10px;
+        }
+    .table tr:nth-child(even) {
+        background-color: #f2f2f2;
+        }
+    .table thead th {
+        position: sticky;
+        top: 0;
+        z-index: 1;
+        }
+    .table thead th:first-child {
+        position: sticky;
+        left: 0;
+        z-index: 2;
+        }
+    .table tbody th {
+        position: sticky;
+        left: 0;
+        z-index: 1;
+        }
+    .table thead tr th:first-child,
+    .table tbody tr td:first-child {
+        width: 300px;
+        min-width: 300px;
+        max-width: 300px;
+        }
+    .title {
+        font-family: Arial, sans-serif;
+        font-size: 32px;
+        font-weight: bold;
+        }
+
+    '''
+    return f'<style>{custom_css}</style>'
+
+
+def _generate_html_report(step, figures):
+    html_report = "<!DOCTYPE html><html lang='en'>"
+    html_report += _generate_html_head(step)
+    html_report += _generate_html_body(figures)
+    html_report += "</html>"
+    return html_report
+
+
+def _generate_html_head(step):
+    head = "<head>"
+    head += f"<title>Analysis report of step {step}</title>"
+    head += f"<p class='title'>Analysis report of step {step}</p>"
+    head += _css_styles_for_report()
+    head += "</head>"
+    return head
 
 
-def report_generator(boxes, scatters, table, step):
+def _generate_html_body(figures):
+    body = "<body>"
+    include_plotlyjs = "cdn"
+    for figure in figures:
+        if isinstance(figure, str):  # tables
+            inner_html = f'''<div class="table">{figure}</div>'''
+        else:  # plots
+            inner_html = figure.to_html(
+                full_html=False, include_plotlyjs=include_plotlyjs
+                )
+            include_plotlyjs = False
+        body += "<br>"  # add a break between tables and plots
+        body += inner_html
+    body += _ngl_viewer()
+    body += "</body>"
+    return body
```

Comment:
could you add a small docstring to this function? explaining what is figures (array of plots, I suppose)

Reply:
added!